### DISCUSSION TITLE
scalar_bar for unstructured grid not working properly

### DISCUSSION THREAD

**Discussion by mirkotriso** [pyvista/pyvista#3652] [Q&A] [answered]
Hello, I have an issue when trying to plot an unstructured grid with associated scalar information as cell_data. I have created a recursive binning algorithm that I used to create an adaptive mesh, which I then save to an unstructured grid in a vtk file. To each cell, I also associate scalar information, e.g., the volume of each cubic cell I generated.

```
ast = pyvista.read(objfile)
ast.scale([1000, 1000, 1000], inplace=True)
mesh = pyvista.read(filename)

p = pyvista.Plotter(notebook=False)
p.add_mesh(ast)
p.add_mesh(mesh, scalars='volume', show_edges=False, show_scalar_bar=True)

p.show()
```

When I try to visualise the data, I get something like the image below, with a very strange looking colorbar and the colors not properly associated to each cell.

<img width="646" alt="image" src="https://user-images.githubusercontent.com/11493001/204140273-a7bd2f95-642f-4b31-9989-0d34d9fc2691.png">

Even stranger is that when I specify `show_edges = True`, the colors of the cells look fine (there are only 3 sizes of cells in this case), while the colorbar still looks strange.

<img width="622" alt="image" src="https://user-images.githubusercontent.com/11493001/204140425-05856b5a-6ee5-4e26-8391-e31c4ff7e3a6.png">

Can anyone help me with this issue?

**Comment by mirkotriso** [comment, marked as answer]
I will give a sort of answer to my own question. Since just one day before I had this problem my code was working fine, I tried to trace back everything I did before running the script again.

It turned out that the cause of this problem was my update to the Intel Iris graphics card drivers. Once I reverted back to the previous version of the drivers the script worked again as expected.

![image](https://user-images.githubusercontent.com/11493001/204176436-204986a9-362d-435d-bbd4-b3e1c39ba21c.png)

I am not sure if this is a bug but it is certainly a behaviour I did not expect, so I thought it was worth mentioning it here.

**Reply by banesullivan** [reply]
Thanks for following up. I was going to say that this looked like a an issue from using an older version of OSMesa (perhaps on Windows). Glad that updating the driver fixd tis for you
